runtime/compiler/runtime/RelocationRuntime.cpp

```diff
@@ -677,6 +735,11 @@ TR_RelocationRuntime::relocateMethodMetaData(UDATA codeRelocationAmount, UDATA d
       _exceptionTable->riData = (void *) (((U_8 *)_exceptionTable->riData) + dataRelocationAmount);
       }
 
+#if defined(JITSERVER_SUPPORT)
+   if (_exceptionTable->osrInfo)
+      _exceptionTable->osrInfo = (void *) (((U_8 *)_exceptionTable->osrInfo) + dataRelocationAmount);
```

Comment:
When the ifdef is going to be lifted this code should work for jitserver and non-jitserver mode. Is `dataRelocationAmount` going to be 0 for non-jitaas?

Reply:
This code is added in 48f04858 for JITServer. Not sure if it'd work for master branch. 

>Is dataRelocationAmount going to be 0 for non-jitaas?

Based the parameter passed into `relocateMethodMetaData`. My guess is no. I'll run some test and see how it goes. 
https://github.com/eclipse/openj9/blob/4f85fc6e42d39e14c044531dd18a7b69b7e41457/runtime/compiler/runtime/RelocationRuntime.cpp#L464-L466

Reply:
If I run the same sanity test in JITServer and non-JITServer mode, they seem to trigger different code path. In non-JITServer mode, `relocateMethodMetaData` is not called. 
Making this change to non-JITServer mode would require extra test in 'OSR`. We could guard it with JITSERVER_SUPPORT and make the change and test later in a separate PR.

Reply:
Looks like for JITServer, `prepareRelocateAOTCodeAndData()` could be called in non-AOT mode to relocate JIT compilations as below. That explain why I don't see `relocateMethodMetaData` being called in non-JITServer mode with the same test case. As per discussion, will guard this code with `JITSERVER_SUPPORT` for now and open a issue to track the future change to the `master` branch.

https://github.com/eclipse/openj9/blob/4f85fc6e42d39e14c044531dd18a7b69b7e41457/runtime/compiler/control/JITClientCompilationThread.cpp#L2481-L2487.